### ISSUE TITLE
"SELECT... OVER (PARTITION BY..." throwing an exception on the parenthesis

### ISSUE THREAD

**Issue by voivoed** [tekartik/sqflite#307]
I have a view created with the following statement:

`CREATE VIEW horse_last_service AS
SELECT abbreviation, name, service_date, horse_id, updates_visit
FROM ( 
SELECT *, ROW_NUMBER() OVER
(PARTITION BY horse_id ORDER BY horse_id ASC, service_date DESC) AS row_num
FROM horse_history)
WHERE row_num = 1;
`

After copying the db file to my assets, when I run my app and try to access the db, I get the following:

`E/SQLiteLog( 1168): (1) near "(": syntax error
E/SQLiteLog( 1168): (11) database corruption at line 117820 of [0c55d17973]
E/SQLiteLog( 1168): (11) malformed database schema (horse_last_service) - near "(": syntax error
E/DefaultDatabaseErrorHandler( 1168): Corruption reported by sqlite on database: /data/user/0/com.example.farrier_app/app_flutter/farrier.db
E/DefaultDatabaseErrorHandler( 1168): deleting the database file: /data/user/0/com.example.farrier_app/app_flutter/farrier.db
E/SQLiteLog( 1168): (1) no such table: settings`

And my database is deleted. The problem happens when I add the "OVER (PARTITON" line to the SQL statement creating the view.

Is there a fix for this? I found a closed issue where the person reports the same thing, the issue was closed but no fix or solution was provided... this is the link:

https://github.com/tekartik/sqflite/issues/120#issuecomment-440795683

I'm coding using VS Code on Windows 10, and using the Android emulator from Android Studio, by the way. Latest versions of Flutter and Dart.

**Comment by alextekartik** [comment]
Thanks for the report.

According to sqlite documentation (https://www.sqlite.org/windowfunctions.html), windows function (including OVER) has been added in sqlite 3.25 which is likely not the version present on Android.

You can check the sqlite version quickly using:

```dart
print(await db.rawQuery('SELECT sqlite_version()'));
```

I'm not familiar with OVER to give you a good workaround.

**Comment by voivoed** [comment]
I checked online and the SQLite versions shipped with Android are indeed older. I don't think I'll be able to use the window functions anyway to keep backward compatibility (otherwise I'd have to have different versions of the DB for each Android version, it would be crazy). I worked around the issue programatically, not the best but oh, well.

Thanks for looking into this and for the plugin, it's invaluable!
